### PR TITLE
DDF for Sonoff SNZB-06P

### PR DESCRIPTION
**Pull request by lhering**
The DDF as discussed in Issue #7369 and the general.xml changes for the light sensor.

### CODE REVIEW

**Comment by Smanar** [comment]
Hello, there is some changes on recent version.
Try changing `"fn": "zcl`" by `"fn": "zcl:attr"`

**Comment by Smanar** [comment]
Still a typo ^^

`"fn": "None"`

It's "none"

**Comment by lhering** [comment]
Yey no errors :)

This is my first Pull Request, is there anything else I have to do?

**Comment by Smanar** [comment]
NoNo it's fine ^^. Now just need to wait for validation.
Just a comment, 

```
          "refresh.interval": 4,
          "read": {
            "fn": "none"
          },
```

Both line are working together, the device normaly use the bind/report you have set at the end, and if and only if this one fail after the period "refresh.interval" deconz make a poll using the "read" line.
So we always use bigger value for "refresh.interval" but in this situation it's useless because you have disable the "read" line.
So for me, you can remove the refresh.interval, or use a bigger one with the "read" fonction.

For the 
```
         "refresh.interval": 30,
          "read": {
            "at": "0x2001",
            "cl": "0xfc11",
            "ep": 1,
            "fn": "zcl:attr"
          },
```

You haven't set bind/report at the end, it's an exotic cluster so perhaps not supported, so it mean every 30s deconz ask for the value.
Not a big problem as the device is powered, but if the bind/report is working better to use it, and set a bigger value for "refresh.interval" as security if it's not working.

```
{
  "bind": "unicast",
  "src.ep": 1,
  "cl": "0xfc11",
  "report": [
    {
      "at": "0x2001",
      "dt": "0x20",
      "min": 1,
      "max": 300,
      "change": "0x00000001"
    }
  ]
}
```

**Comment by manup** [comment]
Hi please also use a JSON formatter like https://jsonformatter.org

**Comment by lhering** [comment]
@Smanar Thanks, I've changed the parts
@manup The format should look better now

File: devices/sonoff/snzb-06p.json [hunk 1]
```diff
@@ -0,0 +1,126 @@
+{
+  "schema": "devcap1.schema.json",
+  "manufacturername": "SONOFF",
+  "modelid": "SNZB-06P",
+  "vendor": "Sonoff",
+  "product": "SNZB-06P",
+  "sleeper": false,
+  "status": "Gold",
+  "path": "/devices/snzb-06p.json",
```
Comment (SwoopX) [inline devices/sonoff/snzb-06p.json:9]: Please remove the `path` stanza, as it is not necessary :)

**Comment by Smanar** [comment]
Lol a new problem
>The refresh interval (300 - 60 = 240) should be greater than the binding max refresh value (300) with a margin of 60 seconds

You have set the same time in the resfresh.interval, and on the bind/report, the device need to use the bind/report and not the refresh, so we set a bigger timing for the refresh.interval.

File: devices/sonoff/snzb-06p.json [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      "bind": "unicast",
+      "src.ep": 1,
+      "cl": "0x0406",
+      "report": [
+        {
+          "at": "0x0000",
+          "dt": "0x18",
+          "min": 1,
+          "max": 300,
+          "change": "0x00000001"
```
Comment (SwoopX) [inline devices/sonoff/snzb-06p.json:107]: As the datatype is discrete here, `"change": "0x00000001"` is superfluous. Therefore, please remove that line and do not forget about the comma one line above 🙂

File: devices/sonoff/snzb-06p.json [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        },
+        {
+          "name": "config/on"
+        },
+        {
+          "name": "config/reachable"
+        },
+        {
+          "name": "state/dark",
+          "refresh.interval": 300,
```
Comment (SwoopX) [inline devices/sonoff/snzb-06p.json:55]: As the error suggests, the interval should be greater than the one you've chosen at the end of the DDF for attribute reporting. As you chose 300 seconds below, a value of 360 is recommended here.

File: devices/sonoff/snzb-06p.json [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        {
+          "name": "attr/swversion"
+        },
+        {
+          "name": "attr/type"
+        },
+        {
+          "name": "attr/uniqueid"
+        },
+        {
```
Comment (SwoopX) [inline devices/sonoff/snzb-06p.json:47]: You might want to consider adding `config/delay` as well, as the device allows setting the interval when it changes from occupied to unoccupied via attribute 0x0020.

An example would be the Philips motion sensor:
https://github.com/dresden-elektronik/deconz-rest-plugin/blob/master/devices/philips/sml001_motion_sensor.json

File: devices/sonoff/snzb-06p.json [hunk 5]
```diff
@@ -0,0 +1,126 @@
+{
+  "schema": "devcap1.schema.json",
+  "manufacturername": "SONOFF",
+  "modelid": "SNZB-06P",
+  "vendor": "Sonoff",
+  "product": "SNZB-06P",
```
Comment (SwoopX) [inline devices/sonoff/snzb-06p.json:6]: You might want to spend 2-3 additional words what kind of device that is 🙂

File: devices/sonoff/snzb-06p.json [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        }
+      ]
+    }
+  ],
+  "bindings": [
+    {
+      "bind": "unicast",
+      "src.ep": 1,
+      "cl": "0x0500"
+    },
```
Comment (SwoopX) [inline devices/sonoff/snzb-06p.json:96]: AS the IAS Zone cluster does not seem to be used in any way, please remove this part.

**Review by SwoopX — CHANGES_REQUESTED** [review]
Almost there, just some small things to amend from my perspective 👍

**Comment by Smanar** [comment]
@lhering there is still somes changes requested.

**Comment by lhering** [comment]
I dont know whats mssing here. Maybe someone could help me what I have to do.

**Comment by Smanar** [comment]
@lhering oups, sorry you are right.
@SwoopX all is ok for you ? If I m right all your requested changes are dones ?

**Comment by SwoopX** [comment]
Yup, looks indeed like my remarks were all addressed. Thanks.

**Review by SwoopX — APPROVED** [review]

**Comment by ThiemeNL** [comment]
@lhering could you please complete the PR? Thank you for your effort.

**Comment by Smanar** [comment]
It's ok, the PR is complete ^^.

**Comment by LordGMM** [comment]
IMG_1101
Is there a chance to add those parameters to the ddf?

I d like to have two off my sensors to be within a longer range

**Comment by Smanar** [comment]
Can try with the GUI, cluster 0x0406, attribute 0x0022
Or add to the DDF

```
        {
          "name": "config/sensitivity",
          "awake": true,
          "refresh.interval": 3600,
          "read": {
            "at": "0x0022",
            "cl": "0x0406",
            "ep": 1,
            "fn": "zcl:attr"
          },
          "parse": {
            "at": "0x0022",
            "cl": "0x0406",
            "ep": 1,
            "eval": "Item.val = Attr.val;",
            "fn": "zcl:attr"
          },
          "write": {
            "at": "0x0022",
            "cl": "0x0406",
            "dt": "0x20",
            "ep": 1,
            "eval": "Item.val",
            "fn": "zcl:attr"
          }
        },
```

**Comment by mantaalex** [comment]
is it possible to set the sensitivity yet in home assistant or deconz app?

**Comment by Smanar** [comment]
The code on my previous post have never be tested/added.

**Pull request merged**